packages/carbon-graphs/src/js/helpers/axis.js

```diff
@@ -127,17 +133,36 @@ const prepareXAxis = (
   width,
   format,
   orientation = AXES_ORIENTATION.X.BOTTOM,
+  lowerLimit,
+  upperLimit,
 ) => {
   let d3Axis = d3.axisBottom(scale);
   if (isXAxisOrientationTop(orientation)) {
     d3Axis = d3.axisTop(scale);
   }
-  d3Axis
-    .ticks(
-      Math.max(width / constants.MAX_TICK_VARIANCE, constants.MIN_TICKS),
-    )
-    .tickValues(processTickValues(tickValues))
-    .tickFormat(format);
+  if (!tickValues || tickValues.length === 0) {
+    d3Axis
+      .ticks(
+        Math.max(width / constants.MAX_TICK_VARIANCE, constants.MIN_TICKS),
+      )
+      .tickValues(processTickValues(tickValues))
+      .tickFormat(format);
+  } else if (hasDateValues(tickValues)) {
+    const filteredTickValues = tickValues.filter((value) => {
+      const date = new Date(value);
+      return date >= lowerLimit && date <= upperLimit;
```

Comment:
Are date, lowerLimit and upperLimit the same type here? Can you do a console.log and post a screenshot of the results here?

Reply:
yes, same type
<img width="1723" alt="Screenshot 2023-06-28 at 11 36 11 AM" src="https://github.com/cerner/terra-graphs/assets/105538517/ff1e9114-00a4-4fa4-a3d0-3dea2e771c28">
<img width="980" alt="Screenshot 2023-06-28 at 11 36 46 AM" src="https://github.com/cerner/terra-graphs/assets/105538517/cf52cbd3-5cee-4d17-a28e-b43334ce6c0c">